lib/compat/wordpress-6.9/terms-query-block-patterns.php

```diff
@@ -0,0 +1,101 @@
+ *
+ * @since 6.9.0
+ * @access private
+ */
+function gutenberg_terms_query_register_block_patterns() {
+	$should_register_core_patterns = get_theme_support( 'core-block-patterns' );
+
+	if ( $should_register_core_patterns ) {
+		$term_query_block_patterns = array(
+			'term-query-cat-descriptions' => array(
+				'title'      => _x( 'Category Descriptions', 'Block pattern title' ),
+				'blockTypes' => array( 'core/terms-query' ),
+				'categories' => array( 'query' ),
+				'content'    => '<!-- wp:terms-query -->
+					<div class="wp-block-terms-query">
+					<!-- wp:term-template {"style":{"spacing":{"blockGap":"var:preset|spacing|40"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch","flexWrap":"nowrap"}} -->
+					<!-- wp:group {"layout":{"type":"flex","orientation":"vertical","flexWrap":"nowrap","justifyContent":"stretch"}} -->
+					<div class="wp-block-group">
+					<!-- wp:heading {"metadata":{"bindings":{"content":{"source":"core/term-data","args":{"key":"name"}}}}} -->
+					<h2 class="wp-block-heading"></h2>
+					<!-- /wp:heading -->
+					<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"core/term-data","args":{"key":"description"}}}}} -->
+					<p></p>
+					<!-- /wp:paragraph -->
+					<!-- wp:buttons -->
+					<div class="wp-block-buttons">
+					<!-- wp:button {"metadata":{"bindings":{"url":{"source":"core/term-data","args":{"key":"link"}}}},"className":"is-style-fill"} -->
+					<div class="wp-block-button is-style-fill"><a class="wp-block-button__link wp-element-button">View posts</a></div>
+					<!-- /wp:button -->
+					</div>
+					<!-- /wp:buttons -->
+					</div>
+					<!-- /wp:group -->
+					<!-- /wp:term-template -->
+					</div>
+					<!-- /wp:terms-query -->',
+			),
+			'term-query-cat-list'         => array(
+				'title'      => _x( 'Category List', 'Block pattern title' ),
+				'blockTypes' => array( 'core/terms-query' ),
+				'categories' => array( 'query' ),
+				'content'    => '<!-- wp:terms-query {"termQuery":{"taxonomy":"category","hierarchical":true}} -->
+					<div class="wp-block-terms-query">
+					<!-- wp:term-template {"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch","flexWrap":"nowrap"}} -->
+					<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"center"}} -->
+					<div class="wp-block-group">
+					<!-- wp:buttons {"layout":{"type":"flex","verticalAlignment":"center"}} -->
+					<div class="wp-block-buttons">
+					<!-- wp:button {"metadata":{"bindings":{"url":{"source":"core/term-data","args":{"key":"link"}},"text":{"source":"core/term-data","args":{"key":"name"}}}},"className":"is-style-outline","style":{"spacing":{"padding":{"left":"0","right":"0","top":"0","bottom":"0"}},"border":{"color":"#00000000","radius":{"topLeft":"0px","topRight":"0px","bottomLeft":"0px","bottomRight":"0px"}}}} -->
```

Comment:
We need to avoid as much of this styling as we can. Colors won't work on all themes, we should allow the attern to inherit whatever the theme provides. I would also avoid using button variations like outline (the other pattern has an is-style-fill class too that we can probably remove too)